src/Leaderboard.ts

```diff
@@ -177,4 +191,8 @@ export default class Leaderboard {
     // update all leaderbaord messages
     for (const msg of this._messages) msg.edit(newMsg).catch(console.error);
   }
+
+  private getDay(now: Date, day: number): Date {
+    return new Date(`${now.getFullYear()}-12-${("0" + day).slice(-2)}T00:00:00-05:00`);
```

Comment:
Instead of “hacking” the day to be max 2 letters you could also just use the function at https://github.com/TrojanerHD/AdventofCode-Bot/blob/48124a750fefc536f6d20926d78701de043a3316/src/ChannelUpdater.ts#L69 to parse your day correctly

Reply:
Maybe move the function to https://github.com/TrojanerHD/AdventofCode-Bot/blob/main/src/DiscordBot.ts, make it publicly available and static? Some changes in the DiscordBot.ts would be necessary but that would be cleaner imo.

Reply:
I don't think DiscordBot.ts is the right file for that, cause, at the name says, it is for the discord client itself, not for common functions

Reply:
maybe I should create a common.ts file, containing these functions?

Reply:
but I think for now this works and mayor reconstruction of the structure should be done in another branch. nor now I think this is the way to go.